pkg/client/src/app/queries/tasks.ts

```diff
@@ -14,24 +14,31 @@ export const useFetchTasks = (
   defaultIsFetching: boolean = false
 ): IFetchState => {
   const [tasks, setTasks] = useState<Array<Task>>([]);
-  const { isLoading, data, error } = useQuery("tasks", () =>
-    getTasks()
-      .then(({ data }) => {
-        let uniqLatestTasks: Task[] = [];
-        data.forEach((task) => {
-          const aTask = uniqLatestTasks.find(
-            (item) => task.application.id === item.application.id
-          );
-          if (aTask && aTask.createTime && task.createTime > aTask.createTime) {
-            const others = uniqLatestTasks.filter((t) => t.id !== aTask.id);
-            uniqLatestTasks = [...others, task];
-          } else uniqLatestTasks.push(task);
-        });
-        setTasks(uniqLatestTasks);
-      })
-      .catch((error) => {
-        console.log("error, ", error);
-      })
+  const { isLoading, data, error } = useQuery(
+    "tasks",
+    () =>
+      getTasks()
+        .then(({ data }) => {
+          let uniqLatestTasks: Task[] = [];
+          data.forEach((task) => {
+            const aTask = uniqLatestTasks.find(
+              (item) => task.application.id === item.application.id
+            );
+            if (
+              aTask &&
+              aTask.createTime &&
+              task.createTime > aTask.createTime
+            ) {
+              const others = uniqLatestTasks.filter((t) => t.id !== aTask.id);
+              uniqLatestTasks = [...others, task];
+            } else uniqLatestTasks.push(task);
+          });
+          setTasks(uniqLatestTasks);
+        })
+        .catch((error) => {
+          console.log("error, ", error);
+        }),
+    { refetchInterval: 1000 * 60 }
```

Comment:
Maybe we want to fetch more often and push it out if we a see a performance hit?

Reply:
Can also disable query if a condition is happening like a wizard modal being open etc. https://react-query.tanstack.com/guides/disabling-queries